explorer/e2e/anvil/anvil-filters.spec.ts

```diff
@@ -0,0 +1,142 @@
+const filter_index_list = [3, 4, 5, 10, 6, 2];
+const filter_index_list_short = [1, 10, 3];
+test("Check that all filters exist on the Datasets tab and are clickable", async ({
+  page,
+}) => {
+  await testFilterPresence(page, anvilTabs.datasets, anvilFilters);
+});
+
+test("Check that all filters exist on the Donors tab and are clickable", async ({
+  page,
+}) => {
+  await testFilterPresence(page, anvilTabs.donors, anvilFilters);
+});
+
+test("Check that all filters exist on the BioSamples tab and are clickable", async ({
+  page,
+}) => {
+  await testFilterPresence(page, anvilTabs.biosamples, anvilFilters);
+});
+
+test("Check that all filters exist on the Activities tab and are clickable", async ({
+  page,
+}) => {
+  await testFilterPresence(page, anvilTabs.activities, anvilFilters);
+});
+
+test("Check that all filters exist on the Files tab and are clickable", async ({
+  page,
+}) => {
+  await testFilterPresence(page, anvilTabs.files, anvilFilters);
+});
+
+test("Check that the first filter on the Datasets tab creates at least one checkbox, and that checking up to the first five does not cause an error and does not cause there to be no entries in the table", async ({
+  page,
+}) => {
+  // Goto the datasets tab
+  await page.goto(anvilTabs.datasets.url);
+  await expect(
+    page.getByRole("tab").getByText(anvilTabs.datasets.tabName)
+  ).toBeVisible();
+
+  // Select a filter
+  await page
+    .getByRole("button")
+    .getByText(
+      filterRegex(anvilFilters[Math.floor(Math.random() * anvilFilters.length)])
+    )
+    .click();
+  // Expect all checkboxes to be unchecked initially and to work properly
+  await expect(page.getByRole("checkbox").first()).toBeVisible();
+  const all_checkboxes = await page.getByRole("checkbox").all();
+  for (let i = 0; i < all_checkboxes.length && i < 5; i++) {
+    const checkbox = all_checkboxes[i];
+    await checkbox.scrollIntoViewIfNeeded();
+    await expect(checkbox).not.toBeChecked();
+    await checkbox.click();
+    await expect(checkbox).toBeChecked();
+  }
+  await page.locator("body").click();
+  await expect(getFirstElementTextLocator(page, 0)).toBeVisible();
+});
+
+test("Check that filter checkboxes are persistent across pages on an arbitrary filter", async ({
+  page,
+}) => {
+  await testFilterPersistence(
+    page,
+    anvilFilters[3],
+    anvilTabTestOrder.map((x) => anvilTabs[x])
+  );
+});
+
+test("Check that filter menu counts match actual counts on the Datasets tab", async ({
+  page,
+}) => {
+  const result = await testFilterCounts(
+    page,
+    anvilTabs.datasets,
+    filter_index_list.map((x) => anvilFilters[x]),
+    25
+  );
+  if (!result) {
+    test.fail();
+  }
+});
+
+test("Check that filter menu counts match actual counts on the Activities tab", async ({
+  page,
+}) => {
+  await testFilterCounts(
+    page,
+    anvilTabs.activities,
+    filter_index_list.map((x) => anvilFilters[x]),
+    25
+  );
+});
+
+test.setTimeout(120000);
```

Comment:
Could you give me the background on why we need the `test.setTimeout` calls?

Reply:
Hi! When I ran these tests locally, I found that they would sometimes take too long and timeout. Adding the setTimeout calls here means the timeout for these particular tests is extended from 60 seconds to 120 seconds, which resolved the issue. I can also reduce the number of tests they run on if that is preferable. If this timeout value was changed in the global config, the timeout for all tests would be longer than necessary, causing tests to take longer if they fail.